index.ts

```diff
@@ -25,7 +26,28 @@ export default function externalizeAllPackagesExcept(noExternals: string[]) {
 					const maybePackageName = getPackageName(args.path);
 					if (!noExternals.includes(maybePackageName)) {
 						try {
-							require.resolve(maybePackageName);
+							// support pnp and yarn berry
+							if (versions.pnp) {
+								const pnpApi = require("pnpapi");
+								for (const locator of pnpApi.getDependencyTreeRoots()) {
+									const locPackage = pnpApi.getPackageInformation(locator);
+									// could be naive implementation not strong checking
+								  	if (locPackage.packageLocation.startsWith(args.resolveDir)) {
+								    		const pkg = locPackage.packageDependencies.get(maybePackageName);
+								    		if (pkg) {
+								      			return { path: args.path, external: true };
+								    		} else {
+								      			throw new Error("not found");
+								    		}
+								  	}
+								}
+							} else {
+								// probably this don't need to be resolved
+								// as it may cause to resolve to transitive deps which is mistake
+								// possible check if package.json includes this dependency may work too
+
+								require.resolve(maybePackageName);
+							}
```

Comment:
could you move this whole thing into a helper function at the bottom of the file? and then we only need to pass it `maybePackageName`

Reply:
Sure, done 👍